src/core/analytics.js

```diff
@@ -789,19 +807,26 @@ class Analytics {
         });
       }
 
-      // convert integrations object to server identified names, kind of hack now!
-      transformToServerNames(rudderElement.message.integrations);
-      rudderElement.message.integrations = getMergedClientSuppliedIntegrations(
-        this.integrationsData,
-        clientSuppliedIntegrations,
-      );
+      // Holding the cloud mode events based on flag and integrations load check
+      if (
+        !options?.bufferDataPlaneEventsUntilReady ||
```

Comment:
This should be a load option and not an event-level option.

Reply:
updated